teamcity: failed tests on master: test/TestReplicateReAddAfterDown

The following tests appear to have failed:

[#50491](https://teamcity.cockroachdb.com/viewLog.html?buildId=50491):

```
--- FAIL: test/TestReplicateReAddAfterDown (0.003s)
	client_test.go:927: [n1,s1,r1/1:/M{in-ax}]: change replicas aborted due to failed preemptive snapshot: breaker open

```
Please assign, take a look and update the issue accordingly.

--- a/pkg/storage/client_raft_test.go
+++ b/pkg/storage/client_raft_test.go
@@ -2363,6 +2363,8 @@ func TestReplicateReAddAfterDown(t *testing.T) {
 	mtc := startMultiTestContext(t, 3)
 	defer mtc.Stop()
 
+	downedStoreIdx := 2
+
 	// First put the range on all three nodes.
 	raftID := roachpb.RangeID(1)
 	mtc.replicateRange(raftID, 1, 2)
@@ -2378,7 +2380,7 @@ func TestReplicateReAddAfterDown(t *testing.T) {
 
 	// Stop node 2; while it is down remove the range from it. Since the node is
 	// down it won't see the removal and clean up its replica.
-	mtc.stopStore(2)
+	mtc.stopStore(downedStoreIdx)
 	mtc.unreplicateRange(raftID, 2)
 
 	// Perform another write.
@@ -2395,8 +2397,8 @@ func TestReplicateReAddAfterDown(t *testing.T) {
 	// replica gets recreated, the replica ID is changed by this
 	// process. An ill-timed GC has been known to cause bugs including
 	// https://github.com/cockroachdb/cockroach/issues/2873.
-	mtc.restartStore(2)
-	mtc.replicateRange(raftID, 2)
+	mtc.restartStore(downedStoreIdx)
+	mtc.replicateRange(raftID, downedStoreIdx)
 
 	// The range should be synced back up.
 	mtc.waitForValues(roachpb.Key("a"), []int64{16, 16, 16})

--- a/pkg/storage/client_test.go
+++ b/pkg/storage/client_test.go
@@ -841,6 +841,18 @@ func (m *multiTestContext) restartStore(i int) {
 	m.senders[i].AddStore(m.stores[i])
 }
 
+// disableCircuitBreakersToStore ensures that no store's RaftTransport will
+// reject requests just because past connections to the given store have failed.
+// If you're stopping store and restarting them, you may want to use this to
+// ensure that errors from when the store was down don't cause other nodes to
+// avoid opening new connections to it.
+func (m *multiTestContext) disableCircuitBreakersToStore(i int) {
+	nodeID := roachpb.NodeID(i + 1)
+	for _, transport := range m.transports {
+		transport.GetCircuitBreaker(nodeID).ShouldTrip = nil
+	}
+}
+
 func (m *multiTestContext) Store(i int) *storage.Store {
 	m.mu.Lock()
 	defer m.mu.Unlock()

--- a/pkg/storage/client_test.go
+++ b/pkg/storage/client_test.go
@@ -36,7 +36,6 @@ import (
 
 	"github.com/cenk/backoff"
 	"github.com/coreos/etcd/raft"
-	"github.com/facebookgo/clock"
 	"github.com/kr/pretty"
 	"github.com/pkg/errors"
 	circuit "github.com/rubyist/circuitbreaker"
@@ -276,22 +275,11 @@ func (m *multiTestContext) Start(t *testing.T, numStores int) {
 	if m.rpcContext == nil {
 		m.rpcContext = rpc.NewContext(log.AmbientContext{}, &base.Config{Insecure: true}, m.clock,
 			m.transportStopper)
-		// Create breaker options which retry very quickly and use a "real"
-		// clock so breakers retry without requiring the manual clock to be
-		// incremented manually to untrip the breaker.
+		// Create a breaker which never trips and never backs off to avoid
+		// introducing timing-based flakes.
 		m.rpcContext.BreakerFactory = func() *circuit.Breaker {
-			b := &backoff.ExponentialBackOff{
-				InitialInterval:     1 * time.Millisecond,
-				RandomizationFactor: 0.25,
-				Multiplier:          2,
-				MaxInterval:         20 * time.Millisecond,
-				MaxElapsedTime:      0,
-				Clock:               clock.New(),
-			}
-			b.Reset()
 			return circuit.NewBreakerWithOptions(&circuit.Options{
-				BackOff:    b,
-				ShouldTrip: circuit.ThresholdTripFunc(1),
+				BackOff: &backoff.ZeroBackOff{},
 			})
 		}
 	}
@@ -839,18 +827,6 @@ func (m *multiTestContext) restartStore(i int) {
 	}
 	// The sender is assumed to still exist.
 	m.senders[i].AddStore(m.stores[i])
-}
-
-// disableCircuitBreakersToStore ensures that no store's RaftTransport will
-// reject requests just because past connections to the given store have failed.
-// If you're stopping store and restarting them, you may want to use this to
-// ensure that errors from when the store was down don't cause other nodes to
-// avoid opening new connections to it.
-func (m *multiTestContext) disableCircuitBreakersToStore(i int) {
-	nodeID := roachpb.NodeID(i + 1)
-	for _, transport := range m.transports {
-		transport.GetCircuitBreaker(nodeID).ShouldTrip = nil
-	}
 }
 
 func (m *multiTestContext) Store(i int) *storage.Store {

